src/main/java/com/liveramp/cascading_ext/jobs/JobUtil.java

```diff
@@ -42,13 +42,24 @@ public static TaskSummary getSummary(JobClient client, JobID id) throws IOExcept
   private static List<TaskFailure> getTaskFailures(JobClient client, JobID id) throws IOException {
     List<TaskFailure> jobFailures = new ArrayList<>();
     RunningJob job = client.getJob(id);
-    TaskCompletionEvent[] events = job.getTaskCompletionEvents(0);
+
+    int index = 0;
+    TaskCompletionEvent[] events = job.getTaskCompletionEvents(index);
     ArrayList<TaskCompletionEvent> failures = new ArrayList<TaskCompletionEvent>();
-    for (TaskCompletionEvent event : events) {
-      if (event.getTaskStatus() == TaskCompletionEvent.Status.FAILED) {
-        failures.add(event);
+
+    //this returns either nothing (if no task exceptions at all) or a subset of the exceptions from the first
+    //index at which exceptions are found in the task completion events
+    while (events.length > 0 && failures.size() == 0)
```

Comment:
hmm.... so this is making me rethink whether we should fetch these on job success.  because by definition we're then going to cycle through every task, in case there are failures, which sounds like it could be expensive for long jobs.